### PR TITLE
🌱 Use a private kcp per disruptive replication test

### LINKED ISSUE #2741: flake: TestReplicationDisruptive
### Describe the bug

```
{Failed  === RUN   TestReplicationDisruptive/TestReplicateShard
=== PAUSE TestReplicationDisruptive/TestReplicateShard
=== CONT  TestReplicationDisruptive/TestReplicateShard
=== CONT  TestReplicationDisruptive/TestReplicateShard
    assertions.go:1691: Waiting for condition, but got: workspace phase is Scheduling, not Ready
    replication_test.go:469: 
        	Error Trace:	util.go:328
        	            				workspaces.go:168
        	            				workspaces.go:200
        	            				workspaces.go:206
        	            				replication_test.go:336
        	            				replication_test.go:225
        	            				replication_test.go:469
        	Error:      	Condition never satisfied
        	Test:       	TestReplicationDisruptive/TestReplicateShard
        	Messages:   	failed to wait for root:organization workspace root:e2e-workspace-frbhb to become ready
    --- FAIL: TestReplicationDisruptive/TestReplicateShard (30.11s)
}
```

```
{Failed  === RUN   TestReplicationDisruptive
=== PAUSE TestReplicationDisruptive
=== CONT  TestReplicationDisruptive
=== CONT  TestReplicationDisruptive
    replication_test.go:453: Saving test artifacts for test "TestReplicationDisruptive" under "/logs/artifacts/TestReplicationDisruptive/4127142301".
I0202 15:30:51.749732   46099 crypto.go:615] "generating new CA for cert" certName="kcp-client-ca" certFile="/tmp/TestReplication4102644121/004/client-ca.crt" keyFile="/tmp/TestReplication4102644121/004/client-ca.key"
    replication_test.go:453: Starting kcp servers...
    kcp.go:512: running: /go/src/github.com/kcp-dev/kcp/bin/kcp start --root-directory /tmp/TestReplicationDisruptive102463000/003/kcp/main --secure-port=37823 --embedded-etcd-client-port=33091 --embedded-etcd-peer-port=44443 --embedded-etcd-wal-size-bytes=5000 --kubeconfig-path=/tmp/TestReplicationDisruptive102463000/003/kcp/main/admin.kubeconfig --feature-gates=CustomResourceValidationExpressions=true --audit-log-path /logs/artifacts/TestReplicationDisruptive/4127142301/artifacts/kcp/main/kcp.audit -v=4 --token-auth-file /tmp/TestReplicationDisruptive102463000/001/auth-tokens.csv --audit-policy-file /tmp/TestReplicationDisruptive102463000/002/audit-policy.yaml
=== CONT  TestReplicationDisruptive
    kcp.go:986: waiting for readiness for server at https://10.130.82.12:37823
=== CONT  TestReplicationDisruptive
    kcp.go:1029: success contacting https://10.130.82.12:37823/readyz
    kcp.go:1029: success contacting https://10.130.82.12:37823/livez
    kcp.go:1007: server at https://10.130.82.12:37823 is ready
    replication_test.go:453: Started kcp servers after 18.024790636s
=== CONT  TestReplicationDisruptive
    kcp.go:499: cleanup: canceling context
    kcp.go:503: cleanup: waiting for shutdownComplete
    kcp.go:507: cleanup: received shutdownComplete
--- FAIL: TestReplicationDisruptive (18.11s)
}
```

### Steps To Reproduce

Seen [here in test-e2e](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/kcp-dev_kcp/2728/pull-ci-kcp-dev-kcp-main-e2e/1621166416558297088)

### Expected Behaviour

Appears to be a test flake

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by ncdc**
Fixes #2741

### CODE REVIEW

**Review by stevekuznetsov — APPROVED** [review]

File: test/e2e/reconciler/cache/replication_test.go [hunk 1]
```diff
@@ -444,27 +444,28 @@ func TestReplication(t *testing.T) {
...
-	require.NoError(t, err)
-	cacheClientRT := ClientRoundTrippersFor(kcpRootShardConfig)
-	cacheKcpClusterDynamicClient, err := kcpdynamic.NewForConfig(cacheClientRT)
-	require.NoError(t, err)
-
 	for _, scenario := range disruptiveScenarios {
 		scenario := scenario
+
 		t.Run(scenario.name, func(t *testing.T) {
+			tokenAuthFile := framework.WriteTokenAuthFile(t)
```
Comment (stevekuznetsov) [inline test/e2e/reconciler/cache/replication_test.go:456]: Do we want these to be parallel?

**Review by stevekuznetsov — CHANGES_REQUESTED** [review]

File: test/e2e/reconciler/cache/replication_test.go (hunk 1 above)
Comment (ncdc) [reply test/e2e/reconciler/cache/replication_test.go:458]: Fixed

**Pull request merged**